redeem/path_planner/PathQueue.h

```diff
@@ -134,23 +133,20 @@ class PathQueue
         // subtract one because the optimizer does touch the last index
         curTime += optimizer.beforePathRemoval(queue, readIndex, (writeIndex + queue.size() - 1) % queue.size());
 
-        Path result(std::move(queue[currentReadIndex]));
-
         readIndex = (readIndex + 1) % queue.size();
         availableSlots++;
 
         if (addPathMightBeBlocking && doesQueueHaveSpace())
         {
-            lock.unlock();
             queueHasSpace.notify_all();
         }
-        else if (availableSlots == queue.size())
+
+        if (availableSlots == queue.size())
         {
-            lock.unlock();
```

Comment:
Removing all these unlock calls is just future-proofing - the mutex will be unlocked when `lock` leaves scope at the end of the function anyway. It's just much easier to reason about the code if we continue to hold the lock.

The possible cost is that if we notify any of these condition variables and doing so causes a thread to wake up quickly enough, it could try to acquire the mutex and find the thread in popPath is still holding it. That isn't a problem for correctness, but it could be a performance issue. I think the simplification is worth it, particularly because this situation is very unlikely on a single-processor machine.